botorch/optim/optimize.py

```diff
@@ -273,7 +276,7 @@ def _optimize_batch_candidates(
         if timeout_sec is not None:
             timeout_sec = (timeout_sec - start_time) / len(batched_ics)
 
-        scipy_kws = {
+        gen_kws = {
```

Comment:
Here the variable is defined as `gen_kws` but line 296 uses `gen_kwargs`

Reply:
thanks! I have fixed this in the underlying PR. https://github.com/pytorch/botorch/pull/1655

Reply:
Regarding your comment in the PR repo about pure categorical spaces. Here is a WIP notebook on using pure categorical spaces. Not this implementation and notebook is still in development
[discrete_mixed_bo_categorical.ipynb.txt](https://github.com/pytorch/botorch/files/10688984/discrete_mixed_bo_categorical.ipynb.txt)